csrc/preseg_passes/allocation_order_inference.cpp

```diff
@@ -15,442 +16,274 @@ namespace nvfuser::preseg_passes {
-    // filtering out reduction because they won't appear in consumer's domain.
-    std::vector<IterDomain*> alloc_domain = TensorDomain::noReductions(
-        constructAllocationDomain(producer, alloc_order_map_.at(producer)));
-    // creating producer to consumer root domain map
-    std::unordered_map<IterDomain*, IterDomain*> p2c_map =
-        PairwiseRootDomainMap(producer, consumer).mapProducerToConsumer();
-    // map alloc_domain to consumer
-    std::transform(
-        alloc_domain.cbegin(),
-        alloc_domain.cend(),
-        alloc_domain.begin(),
-        [&p2c_map](IterDomain* id) { return p2c_map.at(id); });
-    return alloc_domain;
-  }
-
-  // Propagate allocation order from producer to consumer via:
-  // 1. Constructs producer allocation_domain with its allocation order;
-  // 2. Mapping it to consumer's root domain to create alloc_domain;
-  // 3. Compute allocation order of consumer as the permutation between
-  //    alloc_domain and `permutation_ref`.
-  //
-  // Returns true when producer has a recorded allocation order, false
-  // otherwise. This function assumes that all root domain in consumer can be
-  // mapped to producer.
-  bool propagateAllocationOrder(
-      TensorView* producer,
-      TensorView* consumer,
-      const std::vector<IterDomain*>& permutation_ref) {
-    auto iter = alloc_order_map_.find(producer);
-    // early return is producer doesn't have an entry in alloc_order_map_
-    if (iter == alloc_order_map_.end()) {
-      return false;
-    }
-
-    // early termination to propagate empty allocation order
-    if (iter->second.empty()) {
-      alloc_order_map_[consumer] = {};
-      return true;
+// Note: we could be using a simplified logic below,
+// See issue https://github.com/NVIDIA/Fuser/issues/2202
+// 1. we project iter domains from targets' rfactor domain which has an exact
+// map to ref's allocation domain.
+//   mapped_id_vec {ir5[i1], iS7[i2]}
+// 2. remove all projected iter domains from target's rfactor
+// domain:
+//   unmapped_ids_vec {iS3[i3], iS4[i4], iS6[i5], ir8[1]}
+// 3. append mapped_id_vec at the end of unmapped_id_vec.
+//   target_alloc_domain
+//   {iS3[i3], iS4[i4], iS6[i5], ir8[1], ir5[i1], iS7[i2]}
+void AllocationOrderMapping(
+    const IdModel& id_model,
+    TensorView* ref,
+    TensorView* target) {
+  const DisjointSets<Val*>& val_sets =
+      id_model.idGraph(IdMappingMode::EXACT).disjointValSets();
+
+  std::vector<IterDomain*> ref_alloc_domain = ref->getMaybeAllocationDomain();
+  const std::vector<IterDomain*>& target_rfactor_domain =
+      target->getMaybeRFactorDomain();
+
+  // map target rfactor domain into ref's allocation domain
+  std::vector<IterDomain*> mapped_id_vec;
+  std::unordered_set<IterDomain*> mapped_id_set;
+
+  // logic to preserve reduction iter domain in target to WAR issue #2202
+#if true
+  for (auto* ref_id : ref_alloc_domain) {
+    for (auto* id : target_rfactor_domain) {
+      // sharp-edges 0
```

Comment:
This is necessary to pass
```
00:21:19 [  FAILED  ] 2 tests, listed below:
00:21:19 [  FAILED  ] AliasTest.NotAllOutputsAlias_Reduction
00:21:19 [  FAILED  ] ResizeTest.PadScheduler4
```

Alternatively, we can also try to use compute at :: INNSERMOST mapping instead of exact mapping in IdModel. But eventually we need to fix https://github.com/NVIDIA/Fuser/issues/2202